internal/worker/main.go

```diff
@@ -85,46 +85,49 @@ func (w *worker) ProcessPermissions(_ context.Context) error {
 
```

Comment:
Code is okay, but method `ProcessPermissions` is kind of too big, can it be reduced?